### DISCUSSION TITLE
[L9] Why are embedded images being turned into attachments?

### DISCUSSION THREAD

**Discussion by jcmcclorey** [laravel/framework#46585] [Q&A]
Laravel: v9.52.4
PHP: v8.0.28

I recently upgraded from v7 to v9 and now I'm noticing that embedded images in my emails are being treated as regular attachments in most email clients and not actually rendering where they should be. I have tested in Outlook, Gmail and ProtonMail and Outlook is the only one that renders the images correctly, even so, it shows the attachment icon (paperclip) in the UI as if there is an attachment.

Here's how I'm embedding these images which as far as I can tell matches up with the [docs](https://laravel.com/docs/9.x/mail#inline-attachments).

```blade
<tr>
    <td>
        <a href="https://www.mysite.com" target="_blank">
            <img src="{{ $message->embed(public_path('/path/to/email-image.gif')) }}" width=600 height=104 alt="ALT TEXT" style="background-color: #ffffff" />
        </a>
    </td>
</tr>
```

I came across this [article](https://talk.tidbits.com/t/display-issues-with-embedded-images-in-email/20547/5) which seems to imply that I would want the Content-Type to be set to "multipart/related" in the header. I was able to check the header key-value pairs for emails sent to Outlook and the older, L7 emails have Content-Type set to "multipart/related", but in the newer L9 emails it's set to "multipart/mixed". So I believe that supports my conclusion that I need to use "multipart/related".

In L7 I was using the old `build()` `Mailable` method to build my emails and I have tried using the new `headers()`, `envelope()` and `content()` `Mailable` methods, but there aren't any changes to the header key-value pairs which I have tried setting.

For example:

```php
public function headers() {
    return new Headers(
        messageId: 'custom-message-id@example.com',
        references: ['previous-message@example.com'],
        text: [
            'X-Custom-Header' => 'Custom Value',
        ]
    );
}
```

**Comment by rileysiemens** [comment]
I have the same problem.  Did you ever figure out a solution?

**Reply by jcmcclorey** [reply]
Are you using [s-ichikawa/laravel-sendgrid-driver](https://packagist.org/packages/s-ichikawa/laravel-sendgrid-driver)?

**Reply by rileysiemens** [reply]
No i'm using Amazon SES

**Reply by jcmcclorey** [reply]
I did find the issue but it was with this package: [s-ichikawa/laravel-sendgrid-driver](https://packagist.org/packages/s-ichikawa/laravel-sendgrid-driver).

I forked that project and made the tweaks that I needed to. I know it's not what you're using but here are the edits that I made. Hopefully it points you in the right direction. These are attachment settings.

BEFORE
```php
'disposition' => $attachment->getPreparedHeaders()->getHeaderParameter('Parameterized', 'Content-Disposition'),
'content_id' => $attachment->getContentId(),
```

AFTER
```php
'disposition' => $attachment->getPreparedHeaders()->getHeaderParameter('Parameterized', 'Content-Disposition') ?? 'inline',
'content_id' => $this->getAttachmentName($attachment),
```
